scripts/version_buckets.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+"""
+Requires boto3 and click packages to be installed.
+
+1) Create a new virtualenv (python3 -m venv my_venv).
+2) Enable the virtualenv (my_venv/bin/activate).
+3) Install the dependencies: (pip install boto3 click).
+4) Run the command (--help will give you more information).
+
+What does this script do?
+
+Reads all the current buckets, ensures that versioning is switched on for those
+that don't yet have it and sets the life cycle configuration to send
+non-current versions of files to glacier storage after 30 days.
+
+This script should only be run once, when the following PR lands in master in
+the control panel:
+
+https://github.com/ministryofjustice/analytics-platform-control-panel/pull/816
+
+(From that moment on, all S3 buckets will be configured in this way by the
+control panel, as a default.)
+
+This script allows us to preview the potential changes and run the script on
+a defined number of buckets at once.
+"""
+import os
+import click
+import logging
+import sys
+import boto3
+import json
+from botocore.exceptions import ClientError
+from pathlib import Path
+
+
+# A flag to indicate if the command is to be run in verbose mode.
+VERBOSE = False
+
+
+# Setup logging. Logs will appear in user's home directory.
+LOGFILE = os.path.join(str(Path.home()), "bucket_update.log")
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logfile_handler = logging.FileHandler(LOGFILE)
+log_formatter = logging.Formatter("%(levelname)s: %(message)s")
+logfile_handler.setFormatter(log_formatter)
+logger.addHandler(logfile_handler)
+
+
+def check_bucket(name, bucket):
```

Comment:
how about calling this method `is_control_panel_bucket`, to make it clearer what it is checking

Reply:
Done in latest push.